### PR TITLE
Make dartdoc strong-mode clean (#1205).

### PR DESCRIPTION
**Pull request by pq**
Fixes https://github.com/dart-lang/dartdoc/issues/1205

/cc @keertip @devoncarew

### CODE REVIEW

**Comment by pq** [comment]
Will update merged version; please hold. :)

**Comment by pq** [comment]
OK, merge clean and travis happy, @devoncarew, @keertip safe for you to take a look!

File: tool/doc_packages.dart [hunk 1]
```diff
@@ -124,7 +124,7 @@ Future<List<String>> _packageUrls(int page) {
   return http
       .get('https://pub.dartlang.org/packages.json?page=${page}')
       .then((response) {
-    return JSON.decode(response.body)['packages'];
+    return JSON.decode(response.body)['packages'] as List<String>;
```
Comment (devoncarew) [inline tool/doc_packages.dart]: You can also do `new List<String>.from(...)`.

File: tool/doc_packages.dart [hunk 2]
```diff
@@ -133,8 +133,9 @@ Future<List<PackageInfo>> _getPackageInfos(List<String> packageUrls) {
     return http.get(p).then((response) {
       var json = JSON.decode(response.body);
       String name = json['name'];
-      List<Version> versions =
-          json['versions'].map((v) => new Version.parse(v)).toList();
+      List<Version> versions = json['versions']
```
Comment (devoncarew) [inline tool/doc_packages.dart]: You may want to use the new List.from() technique here as well.

**Comment by devoncarew** [comment]
lgtm! the clearer the better

File: tool/doc_packages.dart (hunk 1 above)
Comment (pq) [inline tool/doc_packages.dart]: Good call.

File: tool/doc_packages.dart (hunk 2 above)
Comment (pq) [inline tool/doc_packages.dart]: 👍

**Comment by keertip** [comment]
Looks good!

**Pull request merged**